indico/modules/events/timetable/client/js/reducers.ts

```diff
@@ -115,19 +115,25 @@ export default {
           entry: {startDt},
         } = action;
         const newEntries = {...state.changes[state.currentChangeIdx].entries};
+        const currentDayKey = Object.keys(newEntries).find(key =>
+          newEntries[key].some(e => e.id === entry.id && entry.type === entryType)
+        );
+        let dayEntries = newEntries[currentDayKey];
+        const newDayKey = moment(startDt).format('YYYYMMDD');
 
-        const dayKey = moment(startDt).format('YYYYMMDD');
-        const dayEntries = newEntries[dayKey];
-
-        const editedIndex = dayEntries.findIndex(e => {
-          return e.id === entry.id && e.type === entryType;
-        });
-
-        newEntries[dayKey][editedIndex] = entry;
-
+        if (currentDayKey !== newDayKey) {
+          newEntries[newDayKey].push(entry);
```

Comment:
This is also not immutable. `newEntries` is a new object, but the individual days (arrays) are not. I think we should use something like `newEntries[newDayKey] = [...newEntries[newDayKey], entry]`